src/Host/Client/Impl/Definitions/IRSessionProvider.cs

```diff
@@ -5,19 +5,28 @@
 using System.Collections.Generic;
 using System.Threading;
 using System.Threading.Tasks;
+using Microsoft.R.Host.Client.Host;
 
 namespace Microsoft.R.Host.Client {
     public interface IRSessionProvider : IDisposable {
-        IRSession GetOrCreate(Guid guid);
+        IRSession GetOrCreate(Guid guid, IRHostBrokerConnector brokerConnector);
```

Comment:
```IRHostBrokerConnector``` is passed pretty much to every method. This means that either methods belong to ```IRHostBrokerConnector``` or it is a settable property on ```IRSessionProvider``` or may be imported internally by the implementor of the ```IRSessionProvider```. Can different calls can receive different instances of the ```IRHostBrokerConnector``` ?